app/lib/models/substitution.dart

```diff
@@ -197,13 +222,30 @@ class SubstitutionPlan {
   }
 
   Map<String, dynamic> toJson() => {
-    'days': days.map((d) => d.toJson()).toList(),
-    'lastUpdated': lastUpdated.toIso8601String(),
-  };
+        'days': days.map((d) => d.toJson()).toList(),
+        'lastUpdated': lastUpdated.toIso8601String(),
+      };
 
   SubstitutionPlan.fromJson(Map<String, dynamic> json)
       : days = (json['days'] as List)
-      .map((i) => SubstitutionDay.fromJson(i))
-      .toList(),
+            .map((i) => SubstitutionDay.fromJson(i))
+            .toList(),
         lastUpdated = DateTime.parse(json['lastUpdated']);
 }
+
+/// A data class to store all information for a single substitution day
+class SubstitutionInfo {
```

Comment:
The comment is misleading since, thats what the "SubstitutionDay" class is all about. 
Havent looked into that jet any maybe I am wrong, but if we're rendering HTML in the end anyway, why don't we just feed the renderer the whole div extracted by the parser? Should reduce parsing time, since we do not have to "get the values"

Reply:
We are feeding the HTML directly, just not the table. In the end, it is concatenated. I didn’t have good experiences rendering tables with any HTML widget, as styling, padding, etc., were off, especially with long text.

I think we can look into this in detail once the parsing is fixed, if it’s currently not working for you, to look why it does not work.
(I hope this is not customized per school ^^)